src/main/java/com/conveyal/r5/profile/PerTargetPropagater.java

```diff
@@ -209,19 +220,52 @@ private void invertTravelTimes() {
+     * Repeated propagation to the same target point works, because each propagation call checks whether it reduces
+     * the travel time at each separate iteration. This does mean that different iterations can use different modes,
+     * which matches situations where the rider decides dynamically whether to call a cab or walk depending on the
+     * egress station's proximity to the destination point.
+     *
+     * It is possible that computation would be faster with the iteration order inverted to (mode, targetIndex) instead
+     * of (targetIndex, mode).
+     */
+    private void propagateTransit (int targetIndex) {
+        // All linked pointsets are known to be for the same StreetLayer and PointSet, just different modes.
+        for (LinkedPointSet linkedPointSet : linkedTargets) {
+            propagateTransit(targetIndex, linkedPointSet);
+        }
+    }
+
+    /**
+     * For every "iteration" (departure minute and Monte Carlo schedule), find a complete travel time to the specified
      * target from the given nearby stop, and update the best known time for that iteration and target.
      * Also record the best paths if we're going to be saving transit path details.
      * TODO verify if these are actually travel times (vs. clock times after midnight) and clarify code comments.
      * They appear to be travel times (are compared against cutoffSeconds which is a trip duration).
      */
-    private void propagateTransit (int targetIndex) {
+    private void propagateTransit (int targetIndex, LinkedPointSet linkedTargets) {
 
         // Grab the set of nearby stops for this target, with their distances.
-        TIntIntMap pointToStopLinkageCostTable = targets.pointToStopLinkageCostTables.get(targetIndex);
-
-        StreetRouter.State.RoutingVariable unit = targets.linkageCostUnit;
-
+        TIntIntMap pointToStopLinkageCostTable = linkedTargets.pointToStopLinkageCostTables.get(targetIndex);
+        StreetRouter.State.RoutingVariable unit = linkedTargets.linkageCostUnit;
+
+        /**
+         * Pre-compute and retain a pre-multiplied integer speed to avoid float math in the loop below.
+         * These two variables used to be computed once when the propagator was constructed, but now they must be
+         * computed separately for each egress StreetMode.
+         * At one point we believed float math was slowing down this loop, however it's debatable whether that was due
+         * to casts, the mathematical operations themselves, or the fact that the operations were being completed with
+         * doubles rather than floats.
+         */
+        int speedMillimetersPerSecond = (int) (request.getSpeedForMode(linkedTargets.streetMode) * MM_PER_METER);
+        int egressLegTimeLimitSeconds = request.getMaxTimeSeconds(linkedTargets.streetMode);
+
+        // FIXME: MASSIVE HACK THAT ONLY WORKS ON ONE STUDY IN SWITZERLAND
```

Comment:
Does this comment apply here, or to something that was relegated to an ad-hoc branch?

Reply:
This was initially a massive hack since it had a hard-coded wait time in it. Now, after a change, rather than being a hack I'd say it's implementing one of several possible approaches. Will update the comment.